app/controllers/assessment/grading.rb

```diff
@@ -622,8 +622,8 @@ def load_gradesheet_data
   
     # section filter
     o = params[:section] ? {
-      :conditions => { :assessment_id => id, :users => { :section => @user.section } },
-      :joins => "INNER JOIN users ON submissions.user_id = users.id" # hash doesn't work, Rails = stupid
+      :conditions => { :assessment_id => id, :course_user_data => { :section => @cud.section } },
```

Comment:
I tried this and it complained about there being no course_user_data.section column.  Are you sure this works?